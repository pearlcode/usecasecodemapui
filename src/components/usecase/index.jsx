import Usecase from './usecase';

export default Usecase;
