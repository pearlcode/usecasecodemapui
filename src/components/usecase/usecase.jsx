import React from 'react';
import propTypes from 'prop-types';

const UseCase = ({ usecase }) => (
    <div className="user-case">
usecase
        {usecase.id}
    </div>
);

UseCase.propTypes = {
    usecase: propTypes.shape({}).isRequired,
};

export default UseCase;
