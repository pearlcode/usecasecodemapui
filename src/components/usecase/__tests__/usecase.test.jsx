import {
    shallowWithProps,
} from '../../app/__tests__/base';

import usecase from '../usecase';

import correctData from '../../../__mockdata__/correctUsecases';

import selectById from '../../usecaselist/selectors';

describe('usecase', () => {
    it('should mount', () => {
        const wrapper = shallowWithProps(usecase, {
            usecase: selectById(1, correctData),
        });
        expect(wrapper).toMatchSnapshot();
    });
});
