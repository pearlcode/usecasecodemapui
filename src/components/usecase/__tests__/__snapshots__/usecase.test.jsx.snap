// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usecase should mount 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UseCase
    usecase={
      Object {
        "description": "the first description",
        "id": 1,
        "title": "usecase 1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "usecase",
        1,
      ],
      "className": "user-case",
    },
    "ref": null,
    "rendered": Array [
      "usecase",
      1,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "usecase",
          1,
        ],
        "className": "user-case",
      },
      "ref": null,
      "rendered": Array [
        "usecase",
        1,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
